src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.IVsContainedLanguageFactory.cs

```diff
@@ -48,6 +48,32 @@ private AbstractProject FindMatchingProject(IVsHierarchy hierarchy, uint itemid)
                 }
             }
 
+            // This is just a hack, because the managed project system doesn't provide exensibility for 
+            // use to implement IVsContainedLanguageProjectNameProvider. Assuming they provide us a way to implement
+            // this interface then the existing logic would work as desired without any changes.
+            if (projectName == null)
```

Comment:
Not shown here, I implemented this interface in the Razor codebase, and use it to provide the name of my 'companion' project.

Reply:
As the comments say, this is all just haxx because the project system doesn't open this up for extensibility today. Assuming we worked with them to implement this interface, then method would not need any changes.

Reply:
We're unlikely to do so in the "old" language service for classic MVC and aspx pages, so we would need something there